miden-lib/asm/kernels/transaction/api.masm

```diff
@@ -178,6 +188,87 @@ export.set_account_item
     # => [R', V]
 end
 
+#! Returns VALUE located under specified KEY in map in specified account storage slot. 
+#! Panics if 
+#! - the index is out of bounds (>255).
+#! - the requested storage slot type is not map
+#!
+#! Stack: [index, KEY, ...]
+#! Output: [VALUE, 0]
+#!
+#! - index is the index of the item to get.
+#! - VALUE is the value of the item.
+export.get_account_map_item
+    # check if storage type is map
+    dup exec.account::get_storage_slot_type_info drop
+    # => [slot_type, index, KEY, ...]
+
+    # fails if slot_type is not 1 = map
+    push.STORAGE_MAP eq assert
+    # => [index, KEY, ...]
+    
+    # fetch the account storage item, which is ROOT of the map
+    exec.account::get_item swapw
+    # => [KEY, ROOT ...]
+
+    # fetch the VALUE located under KEY in the tree
+    exec.smt::get 
+    # => [VALUE, ROOT, ...]
+
+    # organize the stack for return
+    swapw dropw
+    # => [VALUE, 0]
+end
+
+#! Inserts specified NEW_VALUE under specified KEY in map in specified account storage slot. 
+#! Panics if 
+#! - the index is out of bounds (>255).
+#! - the requested storage slot type is not map
```

Comment:
should we also mention that it panics if the call does not originate from the account context?

Reply:
Indeed, we should, thanks